app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+from typing import Any
+
+
+# solution
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: Any) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> int:
+        result = 0
+        if isinstance(other, Distance):
+            result = self.km + other.km
+            return Distance(result)
+        if isinstance(other, (int, float)):
+            result = self.km + other
+            return Distance(result)
+
+    def __iadd__(self, other: Any) -> Distance:
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+            return self
+        if isinstance(other, (int, float)):
+            self.km = self.km + other
+            return self
+
+    def __mul__(self: Any, other: Any) -> Distance:
+        if isinstance(other, (int, float)):
+            result = self.km * other
+            return Distance(result)
+
+    def __truediv__(self: Any, other: Any) -> Distance:
```

Comment:
The type hint for `self` should be `Distance` instead of `Any` to be consistent with the class definition.